tools/topology/sof/pipe-waves-codec-demux-playback.m4

```diff
@@ -0,0 +1,176 @@
+# Demux Volume Pipeline with Waves codec
+#
+#  Low Latency Playback with demux and volume.
+#
+# Pipeline Endpoints for connection are :-
+#
+#	Playback Demux
+#	B3 (DAI buffer)
+#
+#
+#  host PCM_P -- B0 --> Waves -> B1 --> volume -- B2 --> Demux(M) -- B3 --> sink DAI0
+#                                                              |
+#                                                          pipeline n+1 --> DAI
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pcm.m4')
+include(`pga.m4')
+include(`muxdemux.m4')
+include(`mixercontrol.m4')
+include(`bytecontrol.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`codec_adapter.m4')
+
+define(`SETUP_PARAMS_NAME', `Waves' `ENDPOINT_NAME' `Setup' PIPELINE_ID)
```

Comment:
I would have put a ifdef(ENDPOINT_NAME) throwing an error if not, so that we don't let m4 proceed in case of errors. Or alternatively if it's not defined set to a default?

Reply:
added an error message in case `ENDPOINT_NAME` is not defined